geetools/ee_image_collection.py

```diff
@@ -1044,12 +1047,32 @@ def reduceInterval(
         # create a reducer from user parameters
         red = getattr(ee.Reducer, reducer)() if isinstance(reducer, str) else reducer
 
+        # to avoid different behaviour between single and multi-band image we hard coded the
+        # band names to be suffixed with the reducer output names.
+        def outputNames(b):
+            return red.getOutputs().map(lambda o: ee.String(b).cat("_").cat(o))
+
+        bandNames = self._obj.first().bandNames().map(outputNames).flatten()
+
+        # original behaviour is keeping in the output the original band names which is not compatible with
+        # muli-output reducer for obvious reasons. We need to change this back to the original band names
+        # suffixed with the reducer output names to avoid non-regression issue we added the
+        # "keep_original_names" parameter that will be removed in downstream version
+        if keep_original_names is True and red.getOutputs().length().getInfo() == 1:
+            msg = (
+                "The `keep_original_names` parameter will be removed in future versions\n"
+                " and band names will not be renamed anymore. To keep the old behaviour, rename manually\n"
+                " the output like `ic = ic.map(lambda i: i.rename([<list of the original bands>]))`.\n"
+                " To drop this warning set the `keep_original_names` parameter to `False`."
+            )
+            warnings.warn(msg, category=DeprecationWarning, stacklevel=2)
+            bandNames = self._obj.first().bandNames()
+
         def reduce(ic):
             ic = ee.ImageCollection(ic)
-            timeList = ic.aggregate_array("system:time_start")
-            start, end = timeList.get(0), timeList.get(-1)
+            start = ic.aggregate_array("system:time_start").get(0)
+            end = ic.aggregate_array("system:time_end").get(-1)
```

Comment:
this is a big change that is not related to the PR's motivation IMO. Is it correct to use `system:time_end`? 🤔

Reply:
That's also something coming from @Canto81 and it makes lots of sense to me. 
in a regular image, time_start and time_end are defining the validity of the image, so when I stich multiple one together the new "validity" interval is between the first "time_start" and the last "time_end". The logic we were applying is making the validty period of the aggregated image short by one step period which I think is wrong. 

Now why I added this modification without a deprecation notice nor a warning is because time_end is vastly under used in GEE context where all the date related prefilter (like filterDate) are only looking at time_start so I'm expecting very little impact on our end user. 

I you think we should stick to the initial mechanism I'll revert it back.

Reply:
Yes, I get it, it makes sense. However, I am not 100% sure `system:time_end` always exists 🤔Also, shouldn't `reduceInterval` be aligned with this change? in that method we are using only `system:time_start`. Sorry, I don't have enough time today to inspect it deeply.

Reply:
I created a dedicated issue: https://github.com/gee-community/geetools/issues/440 to avoid polluting this PR and merge it faster